bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/ChannelUID.java

```diff
@@ -149,4 +149,22 @@ public ThingUID getThingUID() {
         List<String> allSegments = getAllSegments();
         return new ThingUID(allSegments.subList(0, allSegments.size() - 1).toArray(new String[allSegments.size() - 1]));
     }
+
+    /**
+     * Returns the binding id.
+     * 
+     * @return binding id
+     */
+    public String getBindingId() {
+        return getSegment(0);
+    }
+
+    /**
+     * Returns the thing type UID.
+     * 
+     * @return thing type UID
+     */
+    public ThingTypeUID getThingTypeUID() {
+        return new ThingTypeUID(getSegment(0), getSegment(1));
+    }
```

Comment:
I don't think these are needed. You can get them via `getThingUID().getBindingId()`.

Reply:
True, but that involves an extra intermediate object, and is more to type. And to make it uniform, you shouldn't have `getBindingId()` on `ThingUID`, only on `ThingTypeUID`, so this call would then become `getThingUID().getThingTypeUID().getBindingId()`. Quite the mouthful.

Reply:
Since it is possible to construct a ThingUID without thing type, it‘s debatable if we can assume it’s always possible to determine the ThingTypeUID.

Reply:
Maybe you could check if `segment[1]` equals `NO_THING_TYPE` and if this is the case return `null` (or make the return value an `Optional<ThingTypeUID>` and return `Optional.empty()`).

Reply:
You actually cannot infer the thing type from a channel or thing UID. Although it is in most cases the second segment, it is possible that a Thing is migrated to a different type while it keeps its UID. This mechanism has been introduced for Zigbee/ZWave and is in use by a small number of other bindings. Therefore, such a method simply must not exist and there is no correct way to implement it.

Reply:
I didn't think of that (even if I use migrate thing type quite often).